src/imageengine.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Plugin from '../core/plugin.js';
+import buildModelConverter from '../engine/conversion/buildmodelconverter.js';
+import WidgetEngine from './widget/widgetengine.js';
+import { modelToViewImage, viewToModelImage, modelToViewSelection } from './converters.js';
+import { toImageWidget } from './utils.js';
+
+/**
+ * The image engine plugin.
+ * Registers `image` as a block element in document's schema and allows it to have two attributes: `src` and `alt`.
+ * Registers converters for editing and data pipelines.
+ *
+ * @memberof image
+ * @extends core.Plugin.
+ */
+export default class ImageEngine extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ WidgetEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const document = editor.document;
+		const dataPipeline = editor.data;
```

Comment:
`data` and `editing`. That's how we name these variables in most features. Plus, I don't like `document` because it's so deeply anchored in our minds that this is the DOM document that it's causing me problems reading the code. We often call it `doc` therefore.